src/repr/src/adt/datetime.rs

```diff
@@ -470,46 +444,46 @@ impl ParsedDateTime {
     /// - If any component overflows a parameter (i.e. i64).
     pub fn compute_interval(&self) -> Result<Interval, String> {
         use DateTimeField::*;
-        let mut months = 0i64;
-        let mut seconds = 0i64;
-        let mut nanos = 0i64;
+        let mut months = 0i32;
+        let mut days = 0i32;
+        let mut micros = 0i64;
 
         // Add all DateTimeFields, from Millennium to Microseconds.
-        self.add_field(Millennium, &mut months, &mut seconds, &mut nanos)?;
+        self.add_field(Millennium, &mut months, &mut days, &mut micros)?;
 
         for field in Millennium.into_iter().take_while(|f| *f <= Microseconds) {
-            self.add_field(field, &mut months, &mut seconds, &mut nanos)?;
+            self.add_field(field, &mut months, &mut days, &mut micros)?;
         }
 
-        let months: i32 = match months.try_into() {
-            Ok(m) => m,
-            Err(_) => {
-                return Err(format!(
-                    "exceeds min/max months (+2147483647/-2147483648); have {}",
-                    months
-                ))
-            }
-        };
-
-        match Interval::new(months, seconds, nanos.into()) {
+        match Interval::new(months, days, micros) {
             Ok(i) => Ok(i),
             Err(e) => Err(e.to_string()),
         }
     }
     /// Adds the appropriate values from self's ParsedDateTime to `months`,
-    /// `seconds`, and `nanos`. These fields are then appropriate to construct
+    /// `days`, and `micros`. These fields are then appropriate to construct
     /// std::time::Duration, once accounting for their sign.
     ///
     /// # Errors
     /// - If any component overflows a parameter (i.e. i64).
     fn add_field(
         &self,
         d: DateTimeField,
-        months: &mut i64,
-        seconds: &mut i64,
-        nanos: &mut i64,
+        months: &mut i32,
+        days: &mut i32,
+        micros: &mut i64,
     ) -> Result<(), String> {
         use DateTimeField::*;
+        fn div_and_round(i: i128, d: i64) -> Option<i64> {
```

Comment:
It seems strange and imprecise to round the microsecond field every time it's updated, before the precision field is even applied, but I'm pretty sure this is actually the same behavior as postgres: 
https://github.com/postgres/postgres/blob/62cb7427d1e491faf8612a82c2e3711a8cd65422/src/backend/utils/adt/datetime.c#L499-L515

That function is called every time there's a fractional part of some unit. `fsec` there is eventually converted directly to microseconds field and the man pages of `rint` say
> Return the integer nearest X in the direction of the prevailing rounding mode.

Still, I'm curious as to what people think about this.

Reply:
The following test does end up testing this behavior and is consistent with postgres, but I think I can definitely add a couple more
```
query T
SELECT INTERVAL '-1.555555555 years 2.555555555 months -3.555555555 days 4.555555555 hours -5.555555555 minutes 6.555555555 seconds';
----
-1 years -4 months +13 days 07:07:53.220829
```

Reply:
Awesome--as long as this is consistent with PG, despite being unintuitive, seems like the right call